tutorials/radiant-mlhub-on-demand-training-data.ipynb

```diff
@@ -0,0 +1,1281 @@
+    "\n",
+    "    Returns:\n",
+    "        GeoDataFrame object with the .explore() method called\n",
+    "    \"\"\"\n",
+    "    item_feature_collection = items.to_dict()\n",
+    "    geom_df = GeoDataFrame.from_features(item_feature_collection).set_crs(4326)\n",
+    "    return geom_df[[\"geometry\", \"datetime\"]].explore(\n",
+    "        column=\"datetime\", style_kwds={\"fillOpacity\": 0.2}, cmap=\"viridis\"\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ab923855-3c20-4f10-af00-d4175a54fdd4",
+   "metadata": {},
+   "source": [
+    "Here are the BigEarthNet chips with their bounding boxes that matched the spatial parameters for the city of Luxembourg and surrounding areas."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b86bf5d8-8dc8-491d-b3cd-1ea1263774ca",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "explore_search_extent(origin_label_items)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "13c6e607-c79b-4678-a483-9bacb0b3b1df",
+   "metadata": {},
+   "source": [
+    "### Download the entire label collection for BigEarthNet from Radiant MLHub"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2e131160-50bb-487f-8fcb-96d37ce80167",
+   "metadata": {},
+   "source": [
+    "We could certainly use the method above to query label Items directly from our connection to the Radiant MLHub API endpoint. However, on very large collections, such as in the case with BigEarthNet, pagination becomes a bottleneck issue in obtaining and resolving STAC items, as it only returns 100 items at a time.  Querying the entire Collection of nearly ~600,000 Items could take hours.\n",
+    "\n",
+    "Therefore, downloading the label Collection (which is only 160 MB) directly is preferrable to paginating over the entire Collection using the API."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "93f67824-bf8f-4cce-862e-856d9cfed26d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "label_collection_path = os.path.join(\n",
+    "    TMP_DIR, BIGEARTHNET_LABEL_COLLECTION, \"collection.json\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "21bd7e52-6bc7-4a8d-bf56-060e80f4019b",
+   "metadata": {},
+   "source": [
+    "Check if collection folder already exists before downloading 173 mb dataset. Otherwise download and uncompress the `.tar.gz` file to extract the label collection files."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "81f5a8c4-c604-4f8a-8fb9-1d90607206ee",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if not os.path.exists(label_collection_path):\n",
+    "    collection = Collection.fetch(BIGEARTHNET_LABEL_COLLECTION)\n",
+    "    archive_path = collection.download(TMP_DIR)\n",
+    "    !tar -xf {archive_path.as_posix()} -C {TMP_DIR}\n",
```

Comment:
It's unfortunate that this decompression takes so long :/ Any thoughts on if you can operate directly on the compressed `.gz`  file? Probably not.

Reply:
I agree, it is unfortunate how long the decompression takes.

I didn't have a great work around to this, as it is related to your previous comment about using `limit`, and taking a random sample of Items from the larger dataset of 600,000 Items. 

I suppose if we fetch only a few thousand to begin with, pagination shouldn't be an issue then we don't need to deal with the `.tar.gz` file. However that likely won't be a random result, I'm guessing the API will just return the first XXX Items?